src/main/java/teammates/logic/core/FeedbackResponseCommentsLogic.java

```diff
@@ -237,99 +234,148 @@ public boolean isNameVisibleTo(FeedbackResponseCommentAttributes comment, Feedba
-                return true;
-            } else if (type == FeedbackParticipantType.RECEIVER_TEAM_MEMBERS
-                    && responseRecipientTeam.equals(currentUserTeam)) {
-                return true;
-            } else if (type == FeedbackParticipantType.STUDENTS && roster.getStudentForEmail(userEmail) != null) {
-                return true;
-            }
-        }
-        return false;
+        return doesVisibilityRecipientTypeMatchUserDetails(response, userEmail, roster, showNameTo, responseGiverTeam,
+                responseRecipientTeam, currentUserTeam);
     }
     
     /**
      * Verify whether the comment is visible to certain user
+     * Does not take individual instructor privileges into account
      * @return true/false
      */
     public boolean isResponseCommentVisibleForUser(String userEmail, String courseId, UserType.Role role,
-            String section, StudentAttributes student, Set<String> studentsEmailInTeam,
-            FeedbackResponseAttributes response, FeedbackQuestionAttributes relatedQuestion,
-            FeedbackResponseCommentAttributes relatedComment, InstructorAttributes instructor) {
-        if (response == null || relatedQuestion == null) {
+            StudentAttributes student, Set<String> studentsEmailInTeam, FeedbackResponseAttributes relatedResponse,
+            FeedbackQuestionAttributes relatedQuestion, FeedbackResponseCommentAttributes comment) {
+        if (relatedResponse == null || relatedQuestion == null) {
             return false;
         }
         
-        boolean isVisibilityFollowingFeedbackQuestion = relatedComment.isVisibilityFollowingFeedbackQuestion;
+        boolean isVisibilityFollowingFeedbackQuestion = comment.isVisibilityFollowingFeedbackQuestion;
         boolean isVisibleToGiver = isVisibilityFollowingFeedbackQuestion
-                                 || relatedComment.isVisibleTo(FeedbackParticipantType.GIVER);
-        
-        boolean isVisibleResponseComment = false;
-        
+                                 || comment.isVisibleTo(FeedbackParticipantType.GIVER);
         
         boolean userIsInstructor = role == Role.INSTRUCTOR;
         boolean userIsStudent = role == Role.STUDENT;
         
-        boolean userIsInstructorAndRelatedResponseCommentIsVisibleToInstructors =
-                userIsInstructor && isResponseCommentVisibleTo(relatedQuestion, relatedComment,
+        boolean isVisibleToUserAsIndividual = 
+                isResponseCommentVisibleToUserAsIndividual(
+                        userEmail, relatedResponse, relatedQuestion, comment,
+                        isVisibleToGiver, userIsInstructor, userIsStudent);
+        
+        boolean isVisibleToUserAsTeamMember = 
+                isResponseCommentVisibleToUserAsTeamMember(
+                        student, studentsEmailInTeam, relatedResponse,
+                        relatedQuestion, comment, userIsStudent);
```

Comment:
Minor change in behaviour here: the code no longer skips the comparison of booleans in `isResponseCommentVisibleToUserAsTeamMember()` if `isVisibleToUserAsIndividual` is true. This is not expected to change the end result.